### PR TITLE
[Merged by Bors] - Add --tail for reading from the end of a log

### LINKED ISSUE #1216: Add `fluvio consume --tail=X` to consume starting X from the end of a stream
Right now, to consume records starting X from the end of a stream, we use `fluvio consume <topic> --offset="-X"`. We think it would be more ergonomic to switch this up so that:

- `--offset` always takes a positive number, counting from the beginning of the stream, and
- `--tail`, a new command, takes a positive number and counts from the end of the stream

### PR DESCRIPTION
**Pull request by nicholastmosher**
Closes #1216

### CODE REVIEW

File: src/cli/src/consume/mod.rs [hunk 1]
```diff
@@ -51,9 +48,17 @@ pub struct ConsumeOpt {
     #[structopt(short, long)]
     pub key_value: bool,
 
-    /// Offsets can be positive or negative. (Syntax for negative offset: --offset="-1")
-    #[structopt(short, long, value_name = "integer")]
-    pub offset: Option<i64>,
+    /// Consume records starting X from the beginning of the log (default: 0)
+    #[structopt(short = "B", value_name = "integer", conflicts_with_all = &["offset", "tail"])]
+    pub from_beginning: Option<Option<u32>>,
```
Comment (sehz) [inline src/cli/src/consume/mod.rs:53]: why is this option of option?

Comment (nicholastmosher) [reply src/cli/src/consume/mod.rs:53]: There are three states:

- `fluvio consume <topic>` (no `--from-beginning`): consume from the end (corresponds to `None`)
- `fluvio consume <topic> --from-beginning`: consume from the very beginning (corresponds to `Some(None)`)
- `fluvio consume <topic> --from-beginning=10`: consume starting 10 after the beginning (`Some(Some(offset))`)

Comment (nicholastmosher) [reply src/cli/src/consume/mod.rs:53]: Same pattern goes for `--tail`

**Review by sehz — APPROVED** [review]

**Pull request closed without merging**